cluster/cluster.go

```diff
@@ -337,23 +344,55 @@ func (executor *GPDBExecutor) ExecuteLocalCommandWithContext(commandStr string,
 	return string(output), err
 }
 
+// Create a new exec.Command object so we can run it again
+func resetCmd(cmd *exec.Cmd) *exec.Cmd {
+	path := cmd.Path
+	args := cmd.Args
+	return exec.Command(path, args...)
+}
+
+/*
+ * ExecuteClusterCommandWithRetries, but only 1 attempt to keep the previous functionality
+ */
+func (executor *GPDBExecutor) ExecuteClusterCommand(scope Scope, commandList []ShellCommand) *RemoteOutput {
+	return executor.ExecuteClusterCommandWithRetries(scope, commandList, 1, 0)
+}
+
 /*
  * This function just executes all of the commands passed to it in parallel; it
  * doesn't care about the scope of the command except to pass that on to the
  * RemoteOutput after execution.
+ *
+ * It will retry the command up to maxAttempts times
  * TODO: Add batching to prevent bottlenecks when executing in a huge cluster.
  */
-func (executor *GPDBExecutor) ExecuteClusterCommand(scope Scope, commandList []ShellCommand) *RemoteOutput {
+func (executor *GPDBExecutor) ExecuteClusterCommandWithRetries(scope Scope, commandList []ShellCommand, maxAttempts int, retrySleep time.Duration) *RemoteOutput {
 	length := len(commandList)
 	finished := make(chan int)
 	numErrors := 0
 	for i := range commandList {
 		go func(index int) {
+			var (
+				out    []byte
+				err    error
+				stderr bytes.Buffer
+			)
```

Comment:
I'm curious why the `out` and `err` declaration were pulled out of line 385.

Reply:
The code was moved into a for loop so if it had been kept where it was, they wouldn't have been accessible below outside of the for loop.